tests/test_permissions.py

```diff
@@ -0,0 +1,1254 @@
+    ),
+    field_id('OmicsRun', 'organization'): TestData(
+        resource_perm=GET_ORGANIZATION, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('OmicsRun', 'status'): TestData(
+        resource_ignore=IgnoreReason.INTRAMODULE, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('OmicsRun', 'workflow'): TestData(
+        resource_ignore=IgnoreReason.INTRAMODULE, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Organization', 'environments'): TestData(
+        resource_perm=GET_ORGANIZATION, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Organization', 'stats'): TestData(
+        resource_ignore=IgnoreReason.INTRAMODULE, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Organization', 'userRoleInOrganization'): TestData(
+        resource_ignore=IgnoreReason.INTRAMODULE, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Permission', 'type'): TestData(
+        resource_ignore=IgnoreReason.INTRAMODULE, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Query', 'countDeletedNotifications'): TestData(
+        resource_ignore=IgnoreReason.NOTREQUIRED, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Query', 'countReadNotifications'): TestData(
+        resource_ignore=IgnoreReason.NOTREQUIRED, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Query', 'countUnreadNotifications'): TestData(
+        resource_ignore=IgnoreReason.USERLIMITED, tenant_ignore=IgnoreReason.USERLIMITED
+    ),
+    field_id('Query', 'countUpVotes'): TestData(resource_ignore=IgnoreReason.PUBLIC, tenant_ignore=IgnoreReason.PUBLIC),
+    field_id('Query', 'generateEnvironmentAccessToken'): TestData(
+        tenant_perm=MANAGE_ENVIRONMENTS, resource_perm=CREDENTIALS_ENVIRONMENT
+    ),
+    field_id('Query', 'getAttachedMetadataForm'): TestData(
+        resource_ignore=IgnoreReason.NOTREQUIRED, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Query', 'getAuthorSession'): TestData(resource_perm=CREATE_DASHBOARD, tenant_perm=MANAGE_DASHBOARDS),
+    field_id('Query', 'getCDKExecPolicyPresignedUrl'): TestData(
+        resource_perm=LINK_ENVIRONMENT, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Query', 'getConsumptionRolePolicies'): TestData(
+        resource_perm=GET_ENVIRONMENT, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Query', 'getDashboard'): TestData(resource_perm=GET_DASHBOARD, tenant_ignore=IgnoreReason.NOTREQUIRED),
+    field_id('Query', 'getDataPipeline'): TestData(resource_perm=GET_PIPELINE, tenant_ignore=IgnoreReason.NOTREQUIRED),
+    field_id('Query', 'getDataPipelineCredsLinux'): TestData(
+        resource_perm=CREDENTIALS_PIPELINE, tenant_perm=MANAGE_PIPELINES
+    ),
+    field_id('Query', 'getDataset'): TestData(
+        resource_ignore=IgnoreReason.NOTREQUIRED, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),  # TODO Review
```

Comment:
Working on PRs that will change this behavior, but we can merge this first and then I'll update